components/brave_wallet_ui/components/desktop/hardware-wallet-connect/hardware_wallet_connect.tsx

```diff
@@ -117,205 +166,163 @@ export const HardwareWalletConnect = ({
-      })
-        .then((result) => {
-          setAccounts(result)
-        })
-        .catch((error) => {
-          setConnectionError(getErrorMessage(error, selectedAccountType.name))
-          setShowAccountsList(false)
+  const loadMoreAccounts = React.useCallback(async () => {
+    const numberOfAccountsToLoad = totalNumberOfAccounts - accounts.length
+    if (numberOfAccountsToLoad <= 0) {
+      return
+    }
+
+    setIsConnecting(true)
+    const result = await loadAccountsFromDevice({
+      startIndex: accounts.length,
+      count: numberOfAccountsToLoad,
+      scheme: currentHardwareImportScheme,
+      onAuthorized: hideAuthorizeDevice
+    })
+    setIsConnecting(false)
+
+    if (result.payload) {
+      setShowAccountsList(true)
+
+      const newAccounts: Array<Required<AccountFromDevice>> =
+        result.payload.map((acc) => {
+          const alreadyInWallet = !!savedAccounts.find((a) => {
+            return (
+              a.accountId.kind === BraveWallet.AccountKind.kHardware &&
+              a.address === acc.address
+            )
+          })
+          return { ...acc, alreadyInWallet, shouldAddToWallet: false }
         })
-        .finally(() => setIsConnecting(false))
-    },
-    [selectedHardwareWallet, selectedAccountType, filecoinNetwork]
-  )
 
-  const getDefaultAccountName = React.useCallback(
-    (account: BraveWallet.HardwareWalletAccount) => {
-      const index = accounts.findIndex((e) => e.address === account.address)
-      let schemeString
-      switch (selectedDerivationScheme) {
-        case LedgerDerivationPaths.Legacy:
-          schemeString = ' (Legacy)'
-          break
-        default:
-          schemeString = ''
+      setAccounts([...accounts, ...newAccounts])
+    } else {
+      setShowAccountsList(false)
+      if (result.error === 'unauthorized') {
+        setShowAuthorizeDevice(true)
+      } else {
+        setConnectionError(
+          getErrorMessage(result.error, selectedAccountType.name)
+        )
       }
-
-      return index === 0
-        ? `${account.hardwareVendor}${schemeString}`
-        : `${account.hardwareVendor} ${index}${schemeString}`
-    },
-    [accounts, selectedDerivationScheme]
-  )
-
-  const onAddAccounts = React.useCallback(() => {
-    const selectedAccounts = accounts.filter((account) =>
-      selectedDerivationPaths.includes(account.derivationPath)
-    )
-    const renamedSelectedAccounts = selectedAccounts.map((account) => ({
-      ...account,
-      name: getDefaultAccountName(account)
-    }))
-    onAddHardwareAccounts(renamedSelectedAccounts)
+    }
   }, [
+    currentHardwareImportScheme,
+    totalNumberOfAccounts,
+    savedAccounts,
     accounts,
-    selectedDerivationPaths,
-    getDefaultAccountName,
-    onAddHardwareAccounts
+    selectedAccountType.name
   ])
 
+  React.useEffect(() => {
+    loadMoreAccounts()
+  }, [currentHardwareImportScheme, totalNumberOfAccounts, loadMoreAccounts])
```

Comment:
Unneeded dependencies in deps array. `loadMoreAccounts` gets recomputed when `currentHardwareImportScheme` and `totalNumberOfAccounts` values change
```suggestion
  }, [loadMoreAccounts])
```

Reply:
Also, will this effect cause all accounts to be loaded since `loadMoreAccounts` uses `accounts` and therefore is recomputed when it sets the new accounts array value? Is this intentional?